src/main/scala/io/shiftleft/fuzzyc2cpg/passes/astcreation/AstCreator.scala

```diff
@@ -0,0 +1,916 @@
+package io.shiftleft.fuzzyc2cpg.passes.astcreation
+
+import io.shiftleft.fuzzyc2cpg.ast.AstNode
+import io.shiftleft.fuzzyc2cpg.ast.walking.ASTNodeVisitor
+import org.slf4j.LoggerFactory
+import io.shiftleft.codepropertygraph.generated.{EdgeTypes, Operators, nodes}
+import io.shiftleft.codepropertygraph.generated.nodes.NewNode
+import io.shiftleft.fuzzyc2cpg.{Defines, Global}
+import io.shiftleft.fuzzyc2cpg.ast.declarations.{ClassDefStatement, IdentifierDecl}
+import io.shiftleft.fuzzyc2cpg.ast.expressions.{
+  AdditiveExpression,
+  AndExpression,
+  Argument,
+  ArgumentList,
+  ArrayIndexing,
+  AssignmentExpression,
+  BinaryExpression,
+  BitAndExpression,
+  CastExpression,
+  CastTarget,
+  Condition,
+  ConditionalExpression,
+  Constant,
+  DeleteExpression,
+  EqualityExpression,
+  ExclusiveOrExpression,
+  Expression,
+  ForInit,
+  Identifier,
+  InclusiveOrExpression,
+  InitializerList,
+  MemberAccess,
+  MultiplicativeExpression,
+  NewExpression,
+  OrExpression,
+  PostIncDecOperationExpression,
+  PtrMemberAccess,
+  RelationalExpression,
+  ShiftExpression,
+  SizeofOperand,
+  UnaryExpression
+}
+import io.shiftleft.fuzzyc2cpg.ast.functionDef.Template
+import io.shiftleft.fuzzyc2cpg.ast.langc.expressions.{CallExpression, SizeofExpression}
+import io.shiftleft.fuzzyc2cpg.ast.langc.functiondef.{FunctionDef, Parameter}
+import io.shiftleft.fuzzyc2cpg.ast.langc.statements.blockstarters.IfStatement
+import io.shiftleft.fuzzyc2cpg.ast.logical.statements.{BlockStarter, CompoundStatement, Label, Statement}
+import io.shiftleft.fuzzyc2cpg.ast.statements.blockstarters.{CatchList, ForStatement}
+import io.shiftleft.fuzzyc2cpg.ast.statements.jump.{
+  BreakStatement,
+  ContinueStatement,
+  GotoStatement,
+  ReturnStatement,
+  ThrowStatement
+}
+import io.shiftleft.fuzzyc2cpg.ast.statements.{ExpressionStatement, IdentifierDeclStatement}
+import io.shiftleft.fuzzyc2cpg.scope.Scope
+import io.shiftleft.passes.DiffGraph
+import io.shiftleft.proto.cpg.Cpg.{DispatchTypes, EvaluationStrategies}
+
+import scala.jdk.CollectionConverters._
+
+private[astcreation] class AstCreator(diffGraph: DiffGraph.Builder,
```

Comment:
I am surprised that you rewrote AstToCpgConverter completely because i would have expected that you only provide an new implementation of CpgAdapter. This would have saved you quite some time or am I missing something here?

Reply:
Yes, you are missing something. The adapter assumes a builder pattern and that doesn't apply to our case classes. I went that direction first. It means copying over the case classes completely on each "add" operation.